src/v/cluster/rm_stm_types.h

```diff
@@ -194,20 +194,27 @@ struct producer_state_snapshot
     struct finished_request
       : serde::envelope<
           finished_request,
-          serde::version<0>,
+          serde::version<1>,
           serde::compat_version<0>> {
         finished_request() = default;
-        finished_request(int32_t first, int32_t last, kafka::offset last_offset)
+        finished_request(
+          int32_t first,
+          int32_t last,
+          kafka::offset last_offset,
+          model::term_id term = model::term_id{-1})
```

Comment:
nit: make it a required param? (forces callers to look at it I think).

Reply:
will add to a follow up